packages/balances/src/modules/EvmErc20Module.ts

```diff
@@ -98,6 +96,35 @@ export const EvmErc20Module: NewBalanceModule<
   const chainConnector = chainConnectors.evm
   assert(chainConnector, "This module requires an evm chain connector")
 
+  const prepareFetchParameters = async (
+    addressesByToken: AddressesByToken<EvmErc20Token>
+  ): Promise<EvmErc20NetworkParams> => {
+    const tokens = await chaindataProvider.tokensById()
```

Comment:
maybe pass tokensById as parameter and make method synchronous ? would avoid fetching tokens twice when called by suscribeBalances